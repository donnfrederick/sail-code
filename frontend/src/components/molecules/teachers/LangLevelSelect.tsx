import FlatButton from 'components/atoms/teachers/FlatButton'
import { Info } from 'models/teachers'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  info: Info
  register(info: Info): void
}

export default (props: Props) => {
  const { info, register } = props
  const registLangLevel = (event: any) => {
    const value = event.target.textContent
    info.desiredCondition = value
    register(info)
  }
  return (
    <Container>
      {levels.map(text => {
        return (
          <FlatButton
            key={text}
            text={text}
            isSelected={info.desiredCondition === text}
            width={640}
            height={100}
            marginBottom={48}
            withBorder={true}
            onClick={registLangLevel}
          />
        )
      })}
    </Container>
  )
}

const levels = ['日本語が得意な方がいい', '日本語が不得意でも構わない']

const Container = styled.div`
  width: 600px;
  height: 600px;
  margin: 0 auto 48px;
`
