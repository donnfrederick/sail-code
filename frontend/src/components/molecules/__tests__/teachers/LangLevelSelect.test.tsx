import LangLevelSelect from 'components/molecules/teachers/LangLevelSelect'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <LangLevelSelect />', () => {
  const info = {
    email: '',
    password: '',
    name: '',
    gender: '',
    picture: '',
    hobbies: [],
    purposes: [],
    desiredCondition: '',
    introduce: '',
    categories: [],
    tags: [],
    selected_tags: [],
    added_tags: []
  }
  const tree = renderer
    .create(<LangLevelSelect info={info} register={() => null} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
