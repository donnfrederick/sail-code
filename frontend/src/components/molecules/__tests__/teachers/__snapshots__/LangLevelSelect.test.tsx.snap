// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <LangLevelSelect /> 1`] = `
<div
  className="sc-htpNat gPrESo"
>
  <button
    className="sc-bdVaJa kXqNMF"
    color="#138efd"
    data-disabled={false}
    data-selected={false}
    fontSize={36}
    height={100}
    onClick={[Function]}
    width={640}
  >
    日本語が得意な方がいい
  </button>
  <button
    className="sc-bdVaJa kXqNMF"
    color="#138efd"
    data-disabled={false}
    data-selected={false}
    fontSize={36}
    height={100}
    onClick={[Function]}
    width={640}
  >
    日本語が不得意でも構わない
  </button>
</div>
`;
